common4j/src/main/com/microsoft/identity/common/java/commands/DeviceCodeFlowTokenResultCommand.java

```diff
@@ -0,0 +1,122 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.commands;
+
+import com.microsoft.identity.common.java.WarningType;
+import com.microsoft.identity.common.java.commands.parameters.DeviceCodeFlowCommandParameters;
+import com.microsoft.identity.common.java.controllers.BaseController;
+import com.microsoft.identity.common.java.controllers.ExceptionAdapter;
+import com.microsoft.identity.common.java.exception.BaseException;
+import com.microsoft.identity.common.java.exception.ErrorStrings;
+import com.microsoft.identity.common.java.logging.Logger;
+import com.microsoft.identity.common.java.opentelemetry.AttributeName;
+import com.microsoft.identity.common.java.opentelemetry.OTelUtility;
+import com.microsoft.identity.common.java.opentelemetry.SpanName;
+import com.microsoft.identity.common.java.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.java.result.AcquireTokenResult;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Scope;
+import lombok.NonNull;
+
+/**
+ * This command is used in the Device Code Flow (DCF) protocol to acquire token.
+ * Takes in a parameters object containing the  desired access scopes along with authorizationResult {@link AuthorizationResult}
+ * and returns a token result.
+ */
+public class DeviceCodeFlowTokenResultCommand extends TokenCommand{
+    private static final String TAG = DeviceCodeFlowTokenResultCommand.class.getSimpleName();
+
+    private final AuthorizationResult mAuthorizationResult;
+
+    public DeviceCodeFlowTokenResultCommand(@NonNull DeviceCodeFlowCommandParameters parameters,
+                                @NonNull AuthorizationResult authorizationResult,
+                                @NonNull BaseController controller,
+                                @SuppressWarnings(WarningType.rawtype_warning) @NonNull CommandCallback callback,
+                                @NonNull String publicApiId) {
+        super(parameters, controller, callback, publicApiId);
+
+        mAuthorizationResult = authorizationResult;
+    }
+
+    @Override
+    public AcquireTokenResult execute() throws Exception {
+        final String methodTag = TAG + ":execute";
+        Logger.verbose(
+                methodTag,
+                "DeviceCodeFlowTokenResultCommand initiating..."
+        );
+
+        final Span span = OTelUtility.createSpanFromParent(
+                SpanName.AcquireTokenDcf.name(), getParameters().getSpanContext()
+        );
```

Comment:
Have we tested if these spans are collected as expected? If not, I suggest we should check this before merging the PR

Reply:
Tested spans :
![image](https://user-images.githubusercontent.com/107152318/219535663-1be5ad3f-e245-41e4-b876-819f2b2d2f40.png)